packages/yarnpkg-core/sources/semverUtils.ts

```diff
@@ -204,7 +204,7 @@ export function stringifyComparator(comparator: Comparator) {
 }
 
 export function simplifyRanges(ranges: Array<string>) {
-  const parsedRanges = ranges.map(range => validRange(range)!.set.map(comparators => comparators.map(comparator => getComparator(comparator))));
+  const parsedRanges = ranges.map(removeSubsets).map(range => validRange(range)!.set.map(comparators => comparators.map(comparator => getComparator(comparator))));
```

Comment:
Can we rather do a `filter` on the `set` after we already parsed it, rather than the hardcoded `||` split followed by `semver.subset` calls (to avoid parsing the ranges twice)?

Reply:
I considered this, but `semver.subset` only accepts a `string` or a `Range` object.  If I have the parsed range, then I need to compare the members of `Range.set` that look like:

```
[
  [
    Comparator {
      options: {},
      loose: false,
      operator: '>=',
      semver: [SemVer],
      value: '>=1.0.1'
    },
    Comparator {
      options: {},
      loose: false,
      operator: '<',
      semver: [SemVer],
      value: '<1.1.0-0'
    }
  ],
  [
    Comparator {
      options: {},
      loose: false,
      operator: '>=',
      semver: [SemVer],
      value: '>=1.0.2'
    },
    Comparator {
      options: {},
      loose: false,
      operator: '<',
      semver: [SemVer],
      value: '<1.1.0-0'
    }
  ]
]
```

So then I need to either:
- Implement my own version of `subset` (yuck), or...
- Reconstruct a proper range out of the above without any more parsing, which really gets into the guts of `semver`

Either case seems like it would add a lot of complexity for marginal benefit.  Given that I already short-circuit on ranges that don't have `||`, I'm expecting the extra cost here is pretty minimal.

What do you think?